osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs

```diff
@@ -146,6 +147,7 @@ private void addItem(Func<BeatmapInfo> beatmap, RulesetInfo? ruleset = null, IRe
             if (mods != null)
                 AddStep($"set mods to {string.Join(",", mods.Select(m => m.Acronym))}", () => songSelect.AsNonNull().Mods.Value = mods);
 
+            AddUntilStep("wait for ongoing operation to complete", () => !(CurrentScreen as OnlinePlayScreen).ChildrenOfType<OngoingOperationTracker>().Single().InProgress.Value);
```

Comment:
Move this above the `set {ruleset.Name} ruleset` step.

Reply:
Should there be one for each, or is checking only once enough? It's basically the room join operation, right?

Reply:
Checking only once is enough. Not entirely sure exactly which things are considered operations, joining room could be done, changing items is definitely one.